[EDGE] y2024 release notes updates

bbyreddy opened the pull request:
- Converted Edge release notes contents into Changelogs and also updated the Edge build artifact names to remove IOT from them.

bbyreddy committed:
Converted Edge release notes contents into Changelogs and also updated the Edge build artifact names to remove IOT from them.

bbyreddy committed:
Included version dates for Edge 10.18

bbyreddy committed:
Renamed Edge's build artifacts to "edge-operator" and "edge-appliance-vm" and Edge appliance component from "Edge VM Appliance" to "Edge Appliance VM"

BeateRixen commented on content/change-logs/edge/edge-appliance-vm-10-18-0-1-Release-identical-to-10-18.md:
```diff
@@ -10,7 +10,7 @@ component:
+version: 10.18.0.0
 ---
-The {{< product-c8y-iot >}} Edge Appliance VM version of the 2024 release is identical to the version of the 10.18 release. For details on the release see the [10.18 release notes](https://cumulocity.com/releasenotes/release-10-18-0/edge-10-18-0/). You can find the documentation for **{{< product-c8y-iot >}} Edge Appliance VM** at https://cumulocity.com/docs/2024/edge/edge-introduction/.
+The {{< product-c8y-iot >}} Edge Appliance VM version of the 2024 release is identical to the version of the 10.18 release. For details on the release see the [release notes](https://cumulocity.com/docs/2024/change-logs/?component=.component-edge-vm-appliance). You can find the documentation for **{{< product-c8y-iot >}} Edge Appliance VM** at https://cumulocity.com/docs/2024/edge/edge-introducti ...
```
This entry does not make much sense as I already am in the release notes.

bbyreddy replied:
@BeateRixen , do we even require this Announcement?

BeateRixen replied:
At least at some place we should probably mention that Edge 10.18 works with Cumulocity 2024. Do we do that at some other place? Then I think we can remove the announcement, now that we have the release notes here.

bbyreddy replied:
Such a statement wasn't there in the docs earlier as well. In the "Data exchange using data broker" we had a statement that this will only work with the current and previous two releases. But with the CD, we have removed that as well. Anyway, we can think of adding such a compatibility statement in the Introduction/versions sections.

For now, I am removing these announcements for Edge 10.18.

BeateRixen commented on content/change-logs/edge/edge-on-kubernetes-10-18-0-1-Release-identical-to-10-18.md:
```diff
@@ -10,7 +10,7 @@ component:
+version: 10.18.0.0
 ---
-The {{< product-c8y-iot >}} Edge on Kubernetes version of the 2024 release is identical to the version of the 10.18 release. For details on the release see the [10.18 release notes](https://cumulocity.com/releasenotes/release-10-18-0/edge-10-18-0/). You can find the documentation for **{{< product-c8y-iot >}} Edge on Kubernetes** at https://cumulocity.com/docs/2024/edge-kubernetes/k8-edge-introdu ...
+The {{< product-c8y-iot >}} Edge on Kubernetes version of the 2024 release is identical to the version of the 10.18 release. For details on the release see the [release notes](https://cumulocity.com/docs/2024/change-logs/?component=.component-edge-on-kubernetes). You can find the documentation for **{{< product-c8y-iot >}} Edge on Kubernetes** at https://cumulocity.com/docs/2024/edge-kubernetes/k ...
```
Same here, does not make much sense to me to link to where I am.

bbyreddy replied:
@BeateRixen , do we even require this Announcement?

bbyreddy replied:
Such a statement wasn't there in the docs earlier as well. In the "Data exchange using data broker" we had a statement that this will only work with the current and previous two releases. But with the CD, we have removed that as well. Anyway, we can think of adding such a compatibility statement in the Introduction/versions sections.

For now, I am removing these announcements for Edge 10.18.

BeateRixen commented:
@bbyreddy Since the date is not visible anyway, as only versions are shown in the index of the yearly releases, can we give the files a date before the one of 2024.0 so that the entries show up at the bottom of the list, and then label them Edge 10.18.0.0, Edge 10.18.0.1?

bbyreddy committed:
Changed the release dates in the date_settings.toml file for the 10.18.0.0 and 10.18.0.1 releases, so that these versions showup in the Index of the Release Notes at the bottom (before the 2024 releases)

bbyreddy committed:
Release 2024 identical to 10.18 announcement is not required anymore as the Edge 10.18 release announcements showup under the respective versions.

bbyreddy committed:
Fixed incorrect date format.

bbyreddy committed:
Fixed release notes link.

bbyreddy committed:
Update connecting-devices.md

BeateRixen commented:
@bbyreddy Looks good now but still should better have Edge as a label. 
@carlosceia In the date-settings file we need a label for Edge 10.18.0.0 and Edge 10.18.0.1 but it seems that it only accepts numbers. Any idea how we can get around it?

bbyreddy committed:
Deprecation announcement heading updated.

bbyreddy committed:
Update edge-appliance-vm-10-18-0-0-Deprecation-of-cumulocity-linux-agent-included-in-the-cumulocity-edge.md

bbyreddy committed:
Added tickets to the bug fixes and features where possible.

bbyreddy committed:
Update connecting-devices.md

BeateRixen commented on content/change-logs/edge/edge-appliance-vm-10-18-0-0.md:
```diff
@@ -0,0 +1,19 @@
+---
+As CentOS 7 will reach its end-of-life on June 30th, 2024, we are transitioning our support to an alternative Operating System for Edge Appliance users. Starting with this release, the Edge Appliance VM is equipped with [Rocky Linux 8](https://rockylinux.org/). Since the Operating System is changed, the upgrade from earlier versions of Edge entails a side-by-side installation accompanied by data  ...
+
+You can find the documentation for **{{< product-c8y-iot >}} Edge Appliance VM** at https://cumulocity.com/docs/2024/edge/edge-introduction/.
```
Is this sentence still needed?

bbyreddy replied:
Not sure, I can remove it. But this was the structure we had in the old release notes, hence copied.

bbyreddy replied:
Removed

BeateRixen commented on content/change-logs/edge/edge-appliance-vm-10-18-0-0.md:
```diff
@@ -0,0 +1,19 @@
+ticket: CIE-4991
+version: 10.18.0.0
+---
+As CentOS 7 will reach its end-of-life on June 30th, 2024, we are transitioning our support to an alternative Operating System for Edge Appliance users. Starting with this release, the Edge Appliance VM is equipped with [Rocky Linux 8](https://rockylinux.org/). Since the Operating System is changed, the upgrade from earlier versions of Edge entails a side-by-side installation accompanied by data  ...
```
```suggestion
As CentOS 7 will reach its end-of-life on June 30th, 2024, we are transitioning our support to an alternative Operating System for Edge Appliance users. Starting with this release, the Edge Appliance VM is equipped with [Rocky Linux 8](https://rockylinux.org/). Since the Operating System is changed, the upgrade from earlier versions of Edge entails a side-by-side installation accompanied by data migration. For information on upgrading Edge to 10.18, see [Migrating Edge 10.17 to 10.18](/edge/edge-update/#migration-1017-to-1018).
```

BeateRixen commented on content/change-logs/edge/edge-appliance-vm-10-18-0-1-Upgrade-Rocky-to-8-10.md:
```diff
@@ -0,0 +1,21 @@
+ticket: CIE-5234
+version: 10.18.0.1
+---
+This release upgrades the Rocky Linux operating system from version 8.9 to 8.10. For details on updating Edge from version 10.18.0.0 to 10.18.0.1, refer to [Updating Edge](https://cumulocity.com/docs/edge/edge-update/). Due to the inclusion of the new Rocky Linux version in the 10.18.0.1 update archive, the archive size has significantly increased, resulting in a longer update process compared to ...
```
```suggestion
This release upgrades the Rocky Linux operating system from version 8.9 to 8.10. For details on updating Edge from version 10.18.0.0 to 10.18.0.1, refer to [Updating Edge](/edge/edge-update/). Due to the inclusion of the new Rocky Linux version in the 10.18.0.1 update archive, the archive size has significantly increased, resulting in a longer update process compared to previous versions.
```

BeateRixen commented on content/change-logs/edge/edge-appliance-vm-10-18-0-1-Upgrade-Rocky-to-8-10.md:
```diff
@@ -0,0 +1,21 @@
+---
+This release upgrades the Rocky Linux operating system from version 8.9 to 8.10. For details on updating Edge from version 10.18.0.0 to 10.18.0.1, refer to [Updating Edge](https://cumulocity.com/docs/edge/edge-update/). Due to the inclusion of the new Rocky Linux version in the 10.18.0.1 update archive, the archive size has significantly increased, resulting in a longer update process compared to ...
+
+For instructions on upgrading Edge from version 10.17, see [Migrating Edge 10.17 to 10.18](https://cumulocity.com/docs/2024/edge/edge-update/#migration-1017-to-1018).
```
```suggestion
For instructions on upgrading Edge from version 10.17, see [Migrating Edge 10.17 to 10.18](/edge/edge-update/#migration-1017-to-1018).
```

BeateRixen commented on content/change-logs/edge/edge-appliance-vm-10-18-0-1-Upgrade-Rocky-to-8-10.md:
```diff
@@ -0,0 +1,21 @@
+
+For instructions on upgrading Edge from version 10.17, see [Migrating Edge 10.17 to 10.18](https://cumulocity.com/docs/2024/edge/edge-update/#migration-1017-to-1018).
+
+You can find the documentation for **{{< product-c8y-iot >}} Edge Appliance VM** at https://cumulocity.com/docs/2024/edge/edge-introduction/.
```
Same here, probably no longer required

bbyreddy replied:
Removed

BeateRixen commented on content/change-logs/edge/edge-on-kubernetes-10-18-0-0-Metrics-monitoring.md:
```diff
@@ -0,0 +1,19 @@
+---
+Edge deployments on Kubernetes can be monitored utilizing Prometheus, an open-source tool designed for monitoring application states. The Edge operator provides a Prometheus-compatible metrics endpoint, allowing you to track a diverse range of resource usage and availability metrics.
+
+For more information on Edge monitoring, refer to [Monitoring](https://cumulocity.com/docs/2024/edge-kubernetes/edge-operations/#monitoring).
```
```suggestion
For more information on Edge monitoring, refer to [Monitoring](/edge-kubernetes/edge-operations/#monitoring).
```

BeateRixen commented on content/change-logs/edge/edge-on-kubernetes-10-18-0-0-Offline-installation.md:
```diff
@@ -0,0 +1,19 @@
+---
+Edge installation on Kubernetes now offers support for installation in an air-gapped environment. This ensures that all dependencies are accessible locally, eliminating the need for internet access during the installation process.
+
+For detailed instructions, refer to [Installing the Edge operator (offline)](https://cumulocity.com/docs/2024/edge-kubernetes/installing-edge-on-k8/#install-operator-offline).
```
```suggestion
For detailed instructions, refer to [Installing the Edge operator (offline)](/edge-kubernetes/installing-edge-on-k8/#install-operator-offline).
```

BeateRixen commented on content/change-logs/edge/edge-on-kubernetes-10-18-0-0-Remote-software-management.md:
```diff
@@ -0,0 +1,19 @@
+---
+Using the {{< product-c8y-iot >}} software management feature, you can now remotely manage applications, microservices, and extensions installed on Edge.
+
+For detailed instructions, refer to [Managing software remotely](https://cumulocity.com/docs/2024/edge-kubernetes/k8-edge-connecting-edge-to-cloud/#software-management-docs).
```
```suggestion
For detailed instructions, refer to [Managing software remotely](/edge-kubernetes/k8-edge-connecting-edge-to-cloud/#software-management-docs).
```

BeateRixen commented on content/change-logs/edge/edge-on-kubernetes-10-18-0-1-Metrics-monitoring-port.md:
```diff
@@ -0,0 +1,19 @@
+---
+The Edge operator's Prometheus-compatible metrics port has been changed from 8443 to 3443 due to a conflict with another core component.
+
+For more information on Edge monitoring, refer to [Monitoring](https://cumulocity.com/docs/2024/edge-kubernetes/edge-operations/#monitoring).
```
```suggestion
For more information on Edge monitoring, refer to [Monitoring](/edge-kubernetes/edge-operations/#monitoring).
```

bbyreddy replied:
By using the relative URI, does it map to 2024 release doc?

BeateRixen replied:
Yes, it maps to the docs in which you are located. We never add the year to any links in the release branches.

BeateRixen commented on content/change-logs/edge/edge-on-kubernetes-10-18-0-1-Offline-installation.md:
```diff
@@ -0,0 +1,19 @@
+---
+Edge installation on Kubernetes now offers support for installation in an offline or air-gapped environment. This ensures that the required software, Helm charts and Docker images, are available in the disconnected environment through an [Open Container Initiative](https://opencontainers.org/) (OCI) compliant private registry.
+
+For detailed instructions, refer to [Installing the Edge operator (offline)](https://cumulocity.com/docs/2024/edge-kubernetes/installing-edge-on-k8/#install-operator-offline).
```
```suggestion
For detailed instructions, refer to [Installing the Edge operator (offline)](/edge-kubernetes/installing-edge-on-k8/#install-operator-offline).
```

BeateRixen commented on content/edge-kubernetes/k8-connecting-devices-to-edge-bundle/connecting-devices.md:
```diff
@@ -12,7 +12,7 @@ Additionally, Edge offers:
 * OPC UA protocol. OPC UA protocols support through the OPC UA device gateway and OPC UA management service.
 
-* Cloud Fieldbus functionality to collect data from fieldbus devices and remotely manage them. For example, Modbus protocol. With the release of {{< product-c8y-iot >}} Edge version {{< c8y-edge-current-version-alt >}}, we are announcing the deprecation of the {{< company-c8y >}} Linux Agent included in the {{< product-c8y-iot >}} Edge offering. For further details see the **Deprecation of {{< co ...
+* Cloud Fieldbus functionality to collect data from fieldbus devices and remotely manage them. For example, Modbus protocol. With the release of {{< product-c8y-iot >}} Edge version {{< c8y-edge-current-version-alt >}}, we are announcing the deprecation of the {{< product-c8y-iot >}} Linux Agent included in the {{< product-c8y-iot >}} Edge offering. For further details, see the announcement in th ...
```
```suggestion
* Cloud Fieldbus functionality to collect data from fieldbus devices and remotely manage them. For example, Modbus protocol. With the release of {{< product-c8y-iot >}} Edge version {{< c8y-edge-current-version-alt >}}, we are announcing the deprecation of the {{< product-c8y-iot >}} Linux Agent included in the {{< product-c8y-iot >}} Edge offering. For further details, see the announcement in the [{{< c8y-edge-current-version-alt >}} release notes](/change-logs/?productarea=.productarea-edge#edge-appliance-vm-10-18-0-0-Deprecation-of-cumulocity-linux-agent-included-in-the-cumulocity-edge).
```

BeateRixen commented on content/change-logs/edge/edge-on-kubernetes-10-18-0-0.md:
```diff
@@ -0,0 +1,18 @@
+---
+{{< product-c8y-iot >}} IoT Edge on Kubernetes has reached General Availability (GA) and is now ready for deployment in production environments.
+
+You can find the documentation for **{{< product-c8y-iot >}} Edge on Kubernetes** at https://cumulocity.com/docs/2024/edge-kubernetes/k8-edge-introduction/.
```
```suggestion
Refer to the [{{< product-c8y-iot >}} Edge on Kubernetes](/edge-kubernetes/k8-edge-introduction/) documentation.
```

bbyreddy committed:
Apply suggestions from code review

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

bbyreddy committed:
Addressed review comments

bbyreddy committed:
Update edge-appliance-vm-10-18-0-0.md

BeateRixen commented:
> @BeateRixen , I addressed all the review suggestions. If you approve the PR, I can merge it and port the same into 2025 and develop branches. Thanks!

@bbyreddy We first need to find a way to add "Edge" to the label otherwise it is confusing as customers assume that the versions contain the entire release notes for 10.18.00 and 10.18.0.1. I pinged Carlos and asked him to take a look.

bbyreddy commented:
> > @BeateRixen , I addressed all the review suggestions. If you approve the PR, I can merge it and port the same into 2025 and develop branches. Thanks!
> 
> @bbyreddy We first need to find a way to add "Edge" to the label otherwise it is confusing as customers assume that the versions contain the entire release notes for 10.18.00 and 10.18.0.1. I pinged Carlos and asked him to take a look.

@BeateRixen , are you expecting the versions 10.18.0.0 and 10.18.0.1 prefixed with Edge. for example, like Edge 10.18.0.0?

bbyreddy committed:
Merge branch 'release/y2024' into edge/y2024-release-notes-updates

bbyreddy committed:
Update date_settings.toml

bbyreddy committed:
Merge branch 'edge/y2024-release-notes-updates' of https://github.com/Cumulocity-IoT/c8y-docs into edge/y2024-release-notes-updates

bbyreddy committed:
Update date_settings.toml

bbyreddy committed:
Update date_settings.toml

bbyreddy committed:
Update date_settings.toml
